packages/core/src/modules/oob/OutOfBandModule.ts

```diff
@@ -240,6 +240,35 @@ export class OutOfBandModule {
     )
   }
 
+  public async createLegacyConnectionlessInvitation<Message extends AgentMessage>(config: {
+    recordId: string
+    message: Message
+    domain: string
+  }): Promise<{ message: Message; invitationUrl: string }> {
+    // Create keys (and optionally register them at the mediator)
+    const routing = await this.mediationRecipientService.getRouting()
+
+    // Set the service on the message
+    config.message.service = new ServiceDecorator({
```

Comment:
I am not the biggest fan of reassigning a parameter variable like here. Since we pass objects by reference in JavaScript, it will override and could cause unexpected behaviour. I suggest just dereferencing with addition:

```typescript
const messageService = new ServiceDecorator({...})

const foo = {
  ...config,
  message: {...config.message, service: serviceMessage}
} 
```

It might be slightly messy code, but it makes sure we don't edit the params.

Reply:
Hmm it's a class instance, so we'd have to clone it somehow. I'll look into it

Reply:
I understand your point, but couldn't find an easy way to do this. We do this in a lot of places and therefore we should look at fixing this a separate PR. Ignoring for now